src/core/api/border_router_api.cpp

```diff
@@ -53,24 +53,17 @@ otError otBorderRouterGetNetData(otInstance *aInstance, bool aStable, uint8_t *a
 
 otError otBorderRouterAddOnMeshPrefix(otInstance *aInstance, const otBorderRouterConfig *aConfig)
 {
-    otError                                error    = OT_ERROR_NONE;
+    otError                                error;
     Instance &                             instance = *static_cast<Instance *>(aInstance);
     const NetworkData::OnMeshPrefixConfig *config   = static_cast<const NetworkData::OnMeshPrefixConfig *>(aConfig);
 
     OT_ASSERT(aConfig != nullptr);
-    // Add Prefix validation check:
-    // Thread 1.1 Specification 5.13.2 says
-    // "A valid prefix MUST NOT allow both DHCPv6 and SLAAC for address configuration"
-    VerifyOrExit(!aConfig->mDhcp || !aConfig->mSlaac, error = OT_ERROR_INVALID_ARGS);
-    // RFC 4944 Section 6 says:
-    // An IPv6 address prefix used for stateless autoconfiguration [RFC4862]
-    // of an IEEE 802.15.4 interface MUST have a length of 64 bits.
-    VerifyOrExit(!aConfig->mSlaac || aConfig->mPrefix.mLength == OT_IP6_PREFIX_BITSIZE, error = OT_ERROR_INVALID_ARGS);
-
-    error = instance.Get<NetworkData::Local>().AddOnMeshPrefix(*config);
+
+    SuccessOrExit(error = instance.Get<NetworkData::Local>().AddOnMeshPrefix(*config));
+
 #if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
-    // Only try to configure Domain Prefix after the parameter is vaidated via above `AddOnMeshPrefix()`.
-    if (error == OT_ERROR_NONE && aConfig->mDp)
+    // Only try to configure Domain Prefix after the parameter is validated via above `AddOnMeshPrefix()`.
```

Comment:
why not put these code to `instance.Get<NetworkData::Local>().AddOnMeshPrefix` too?
I think we are trying to make sure that `otBorderRouterAddOnMeshPrefix` does the exact same thing as  `Get<NetworkData::Local>().AddOnMeshPrefix`, aren't we?

Reply:
Ditto for `otBorderRouterRemoveOnMeshPrefix`

Reply:
Sure, but that may require a significant refactoring of current backbone router implementation, because AddDomainPrefix and AddOnMeshPrefix will call each other recursively if we make the change. I am currently not sure about the best way to do the refactoring, may leave it to another PR?

Reply:
Fine to leave it for another PR. 
Thanks

Reply:
@librasungirl Any comments on moving Domain Prefix related logic to `Get<NetworkData::Local>().AddOnMeshPrefix` or `Get<NetworkData::Local>().RemoveOnMeshPrefix`?

Reply:
@simonlingoogle sgtm